Vizzuality/landgriffon
# Feature LANDGRIF-899 ranking include scenario data

### General description

This PR adds a conditional scenarioId for both Impact Table and Impact Chart, so that elements present in a scenario can be retrieved mixed with the actual data in both cases

Adds a little refactor for sanitising some DTO and minor fixes 
Deprecates some methods in sourcing record servide and repository, and adds a impact repository
as first step to refactoring all data layer access regarding impact data retrieval to that repository

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

Create an Intervention replacing some noticeable stuff (Material, Supplier, Region..). 
Then GET the impact map adding the scenarioId (the FE is not ready to show this stuff properly)
Then you should see the elements you replaced (depending on the groupBy) being retrieved in the response
with a populated value property, in case your specific intervention has had any impact

- Apart from the added feature / bug fix, check overall performance, styling...

## Checklist before merging
